packages/ra-core/src/controller/create/useCreateContext.tsx

```diff
@@ -24,7 +24,7 @@ import { CreateControllerResult } from './useCreateController';
  */
 export const useCreateContext = <RecordType extends RaRecord = RaRecord>(
     props?: Partial<CreateControllerResult<RecordType>>
-): Partial<CreateControllerResult<RecordType>> => {
```

Comment:
record does not need an `id` here though

Reply:
`CreateControllerResult` already does that job with `record?: Partial<RecordType>`:

https://github.com/marmelab/react-admin/blob/f99917435abd73ece4b38f75420028e81ff578c9/packages/ra-core/src/controller/create/useCreateController.ts#L195-L206